Common/src/Common.Infrastructure/IServiceCollectionExtension.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Enmeshed.Common.Infrastructure.Persistence.Repository;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Enmeshed.Common.Infrastructure.Persistence.Context;
+
+namespace Enmeshed.Common.Infrastructure;
+public static class IServiceCollectionExtension
+{
+    public static IServiceCollection AddCommonRepositories(this IServiceCollection services, IConfiguration configuration)
+    {
+        services.AddTransient<IMetricStatusesRepository, MetricStatusesRepository>();
+        services.AddSingleton(new DapperContext(configuration));
```

Comment:
Instead of passing the whole configuration, you should use the Options pattern instead.

```suggestion
        services.Configure<MetricStatusesDapperContextOptions>(options => {
            options.ConnectionString = configuration.GetSection("Modules:Quotas:Infrastructure:SqlDatabase");
        })
        services.AddSingleton<DapperContext>();
```

Or even better, add a parameter that allows to pass the options configuration callback to the extension method, so that it is possible to configure it from the ConsumerApi